torch_geometric/distributed/rpc.py

```diff
@@ -0,0 +1,212 @@
+import atexit
+import collections
+import functools
+import logging
+import threading
+from abc import ABC, abstractmethod
+from typing import Dict, List, Set
+
+from torch._C._distributed_rpc import _is_current_rpc_agent_set
+from torch.distributed import rpc
+
+from .dist_context import DistRole, get_context
+
+_rpc_init_lock = threading.RLock()
+
+_rpc_inited: bool = False
+r""" State of rpc initialization."""
+
+_rpc_master_addr: str = None
+_rpc_master_port: int = None
+
+_rpc_worker_names: Dict[DistRole, List[str]] = None
+
+
+def rpc_is_initialized():
+    return _is_current_rpc_agent_set()
+
+
+def _require_initialized(func):
+    return rpc.api._require_initialized(func)
+
+
+@_require_initialized
+def get_rpc_master_addr():
+    return _rpc_master_addr
+
+
+@_require_initialized
+def get_rpc_master_port():
+    return _rpc_master_port
+
+
+@_require_initialized
+def get_rpc_worker_names() -> Dict[DistRole, List[str]]:
+    return _rpc_worker_names
+
+
+## All gather objects from all role groups.
+
+
+@_require_initialized
+def global_all_gather(obj, timeout=None):
+    r""" Gathers objects from all groups in a list."""
+    if timeout is None:
+        return rpc.api._all_gather(obj)
+    return rpc.api._all_gather(obj, timeout=timeout)
+
+
+@_require_initialized
+def global_barrier(timeout=None):
+    r""" Block until all local and remote RPC processes. """
+    try:
+        global_all_gather(obj=None, timeout=timeout)
+    except RuntimeError as ex:
+        logging.error("Failed to respond to global_barrier")
+
+
+## RPC initialization and shutdown
+
+
+def init_rpc(master_addr: str, master_port: int, num_rpc_threads: int = 16,
+             rpc_timeout: float = 240):
+
+    with _rpc_init_lock:
+        if rpc_is_initialized() is True:
+            return
+        if rpc_is_initialized() is None:
+            raise RuntimeError(
+                "'init_rpc': error to re-init rpc after shutdown.")
```

Comment:
When does this happen? https://github.com/pytorch/pytorch/blob/main/torch/_C/_distributed_rpc.pyi#L123C5-L123C30 indicates that this function always returns `bool`

Reply:
+1